packages/vitest/src/node/core.ts

```diff
@@ -492,6 +491,8 @@ export class Vitest {
       this.invalidates.clear()
       this.snapshot.clear()
       this.state.clearErrors()
+      await this.coverageProvider?.clean(isFirstRun ? this.config.coverage.clean : this.config.coverage.cleanOnRerun)
```

Comment:
Moving `this.coverageProvider.clean()` inside core's `runningPromise` makes sure another test is not started if earlier is still running the cleanup.

Reply:
This makes sense to me. I guess `runningPromise` is like "transaction" to ensure this code path won't be executed in parallel.

With such analogy, isn't it also possible to put `reportCoverage(...)` inside `runningPromise`? This could be simpler than letting `CoverageProvider` manage its own `runningPromise + cancelRun`, but it would also mean that next "re-run" has to wait for the last run's `reportCoverage` processing, which is already known to be stale.

Reply:
That's good point. 🤔 

Maybe it would be best if `reportCoverage` was only called here before the `runningPromise` is reset. 

https://github.com/vitest-dev/vitest/blob/cfd06118730098b9a15540d609d2fa2b598c866e/packages/vitest/src/node/core.ts#L512-L517

Reply:
Right, there is already `this.report('onFinished', ...)`, so it makes it more reasonable to have `reportCoverage` together unless I'm missing something.

Reply:
Changed to this approached. No code changes needed in coverage provider now 🤩